.github/workflows/test-and-build.yml

```diff
@@ -51,3 +51,36 @@ jobs:
 
             - name: Run unit tests
               run: pnpm test
+
+            # AI Agent Tests to run only after file changes
+            - name: Check for AI files changes
```

Comment:
I think it'll be good to extract this out of test and build and have a separate workflow. wdyt?

Reply:
scrap that... I think this triggers it? and does not actually run the test?

Reply:
I thought of separating this into a different workflow, and I started playing around with it:

- build workflow -> creates artifact with build
  - test workflow -> runs all unit tests
  - test ai workflow -> runs AI tests if needed

It has the benefit of separation of concerns and avoids making this file huge. In terms of performance, it only allows us to run the unit + AI tests in parallel, but don't think it gains us much, as the build process is the the one that takes the longest

Reply:
gotcha, thanks for the explanation 👍